numpy/core/fromnumeric.py

```diff
@@ -649,6 +649,49 @@ def transpose(a, axes=None):
     >>> np.transpose(x, (1, 0, 2)).shape
     (2, 1, 3)
 
+    >>> x = np.arange(24).reshape(2, 2, 2, 3)
+    >>> x.shape
+    (2, 2, 2, 3)
+
+    >>> x 
+    array([[[[ 0,  1,  2],
+             [ 3,  4,  5]],
+
+            [[ 6,  7,  8],
+             [ 9, 10, 11]]],
+
+
+           [[[12, 13, 14],
+             [15, 16, 17]],
+
+            [[18, 19, 20],
+             [21, 22, 23]]]])
+
+    >>> x_transposed = np.transpose(x)
+    >>> x_transposed.shape
+    (3, 2, 2, 2)
+
+    >>> x_transposed
+    array([[[[ 0, 12],
+             [ 6, 18]],
+
+            [[ 3, 15],
+             [ 9, 21]]],
+
+
+           [[[ 1, 13],
+             [ 7, 19]],
+
+            [[ 4, 16],
+             [10, 22]]],
+
+
+           [[[ 2, 14],
+             [ 8, 20]],
+
+            [[ 5, 17],
+             [11, 23]]]])
```

Comment:
Showing the arrays is just noise here IMO. This would be clearer as:

```suggestion
    >>> np.transpose(x).shape
    (3, 2, 2, 2)
```